include/circt/Dialect/FIRRTL/FIRRTLStructure.td

```diff
@@ -18,13 +18,14 @@ def CircuitOp : FIRRTLOp<"circuit",
     The "firrtl.circuit" operation represents an overall Verilog circuit,
     containing a list of modules.
   }];
-  let arguments = (ins StrAttr:$name);
+  let arguments = (ins StrAttr:$name, OptionalAttr<ArrayAttr>:$annotations);
```

Comment:
Instead of making the annotations array optional, did you consider representing this state with an empty array?  That reduces the encoding ambiguity between `Some([])` and `None`, and will reduce some annoyance dealing with the optional.  It would be important not to print the empty attribute array in the asmsyntax tho

Reply:
Good idea. This makes more sense with the representation you suggest. I tried this and it proved weird where tblgen wasn't cooperative. I'm going to look at this tomorrow and see if I can get it to work. If it looks like an upstream issue, I'll see about a fix, but may opt to merge this as is and correct later.

Reply:
I'll look at an upstream fix for this. `ArrayAttr` doesn't implement the `constBuilderCall` method which is needed to generate a `DefaultValuedAttr`. There's a kludge to get around this added in 06b2fe4b6aaacc6c74d9c3c8700bdff283765192.